From c2f2c6f953b5fe8d7caadc65fc2b06d420a620be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?George=20Florea=20B=C4=83nu=C8=99?= <georgefb899@gmail.com>
Date: Fri, 12 Apr 2024 02:09:39 +0300
Subject: [PATCH] fix build with ffmpeg 7

---
 filters/f_lavfi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/filters/f_lavfi.c b/filters/f_lavfi.c
index fe7d3e4527..0978408a6b 100644
--- a/filters/f_lavfi.c
+++ b/filters/f_lavfi.c
@@ -1029,7 +1029,11 @@ static const char *get_avopt_type_name(enum AVOptionType type)
     case AV_OPT_TYPE_VIDEO_RATE:        return "fps";
     case AV_OPT_TYPE_DURATION:          return "duration";
     case AV_OPT_TYPE_COLOR:             return "color";
+#if LIBAVUTIL_VERSION_MAJOR < 59
     case AV_OPT_TYPE_CHANNEL_LAYOUT:    return "channellayout";
+#else
+    case AV_OPT_TYPE_CHLAYOUT:          return "channellayout";
+#endif
     case AV_OPT_TYPE_BOOL:              return "bool";
     case AV_OPT_TYPE_CONST: // fallthrough
     default:
-- 
2.44.0

