diff --git a/src/models/tracksmodel.cpp b/src/models/tracksmodel.cpp
index 7a6e69f849b7d5a97c9eec2f4fcef3538c398588..d679f9b4248f0d06cf3ea0ee93c08194dad9bffb 100644
--- a/src/models/tracksmodel.cpp
+++ b/src/models/tracksmodel.cpp
@@ -73,7 +73,9 @@ QHash<int, QByteArray> TracksModel::roleNames() const
 
 void TracksModel::clear()
 {
+    beginResetModel();
     m_tracks.clear();
+    endResetModel();
 }
 
 void TracksModel::addTrack(Track track)
